main.go

```diff
@@ -17,96 +19,168 @@ import (
+	result, err := secretsClient.GetSecretValue(ctx,
+		&secretsmanager.GetSecretValueInput{
+			SecretId: aws.String(secretID),
+		})
+	if err != nil {
+		return nil, wrapErrGetSecret(fmt.Sprintf("unable to get secret %q from SecretsManager: %e", secretID, err))
+	}
 
-	caller_identity, err := sts_svc.GetCallerIdentity(context.TODO(), &sts.GetCallerIdentityInput{})
+	var secretsJSON map[string]any
+	err = json.Unmarshal([]byte(*result.SecretString), &secretsJSON)
 	if err != nil {
-		log.Fatal(err.Error())
+		return nil, wrapErrGetSecret(fmt.Sprintf("unable unmarshal %q JSON from SecretManager: %e", secretID, err))
+	}
+
+	var secrets []string //nolint:prealloc
+	for k, v := range secretsJSON {
+		slog.Info(">>> finding secret value for:", slog.String("secret name", k))
+		secrets = append(secrets, fmt.Sprintf("%s=%s", k, v))
 	}
 
-	var role_arn string = *caller_identity.Arn
+	return secrets, nil
+}
 
-	split_role := strings.Split(role_arn, "/")
+func getRole(ctx context.Context, stsClient stsClient) (string, error) {
+	callerID, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return "", wrapErrGetRole(fmt.Sprintf("unable to get STS caller ID: %e", err))
+	}
 
-	var role_name string = split_role[1]
+	roleParts := strings.Split(*callerID.Arn, "/")
+	if len(roleParts) < 2 || roleParts[1] == "" || roleParts[1] == "*" {
+		return "", wrapErrGetRole(fmt.Sprintf("unable to determine role name from arn: %s", *callerID.Arn))
+	}
 
-	return role_name
+	return roleParts[1], nil
 }
 
-func main() {
-	var region_config config.LoadOptionsFunc
+func getTags(ctx context.Context, iamClient iamClient, role string) ([]string, error) {
+	slog.Info(">>> getting prefixed tags for role:", slog.String("prefix", secretTagPrefix), slog.String("role", role))
+	resp, err := iamClient.ListRoleTags(ctx,
+		&iam.ListRoleTagsInput{
+			RoleName: &role,
+		})
+	if err != nil {
+		return nil, wrapErrGetTags(fmt.Sprintf("unable to get tags for role: %s", role))
+	}
 
-	manual_region, exists := os.LookupEnv("FS_REGION")
-	if exists {
-		region_config = config.WithRegion(manual_region)
-	} else {
-		region_config = config.WithRegion("")
+	var tags []string
+	for _, t := range resp.Tags {
+		tag := *t.Key
+		if strings.HasPrefix(tag, secretTagPrefix) {
+			tags = append(tags, tag)
+		}
 	}
 
-	cfg, err := config.LoadDefaultConfig(context.TODO(), region_config)
+	return tags, nil
+}
+
+func main() {
+	slog.SetDefault(slog.New(slog.NewJSONHandler(os.Stdout, nil)))
+	slog.Info("fetch-secrets starting >>>")
+
+	path, err := exec.LookPath(os.Args[1])
 	if err != nil {
-		log.Fatal(err)
+		slog.Error("executable not found:", slog.Any("error", err))
+		syscall.Exit(1)
 	}
-	iam_svc := iam.NewFromConfig(cfg)
 
-	var role_name = get_role_name(cfg)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(os.Getenv("FS_REGION")))
```

Comment:
There's nothing in the README/Confluence doc around the usage of this env var. Also might be worth pointing out in code here, the behaviour of [`WithRegion`](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/config#WithRegion) when the given value is the empty string.